frontend/app/login.tsx

```diff
@@ -1,12 +1,233 @@
-import { Text, View } from "react-native";
-import React from "react";
+import React, { useState } from "react";
+import { useRouter } from "expo-router";
+import {
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  StyleSheet,
+} from "react-native";
+import axios from "axios";
 
-const Register = () => {
-  return (
-    <View>
-      <Text>Register</Text>
+import * as SecureStore from "expo-secure-store";
+
+async function save(key: string, value: string) {
+  await SecureStore.setItemAsync(key, value);
+}
+
+function LoginPage() {
+  const router = useRouter();
+  const [isSignUp, setSignUp] = useState(false);
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+  const [phone, setPhone] = useState("");
+  const [toggle, setToggle] = useState(false);
+  const [key, onChangeKey] = React.useState("authtoken");
+  const [value, onChangeValue] = React.useState("Your value here");
+
+  const handleSignUp = async () => {
+    //alert("name: " + username + " password: " + password + " email: " + email + " phone: " + phone);
+
+    try {
+      if (
+        name.length > 0 &&
```

Comment:
In practice you would normally use some kind of form validator like zod. I think for our demo this is fine though

Reply:
Given the time constraints, I felt that it was better to just get a working login, but thank you in the future i will make  the login requirements stronger